compiler/rustc_ast/src/tokenstream.rs

```diff
@@ -213,14 +213,18 @@ impl AttrTokenStream {
                         .into_iter()
                 }
                 AttrTokenTree::Attributes(data) => {
-                    let mut outer_attrs = Vec::new();
-                    let mut inner_attrs = Vec::new();
-                    for attr in &data.attrs {
-                        match attr.style {
-                            crate::AttrStyle::Outer => outer_attrs.push(attr),
-                            crate::AttrStyle::Inner => inner_attrs.push(attr),
-                        }
-                    }
+                    // let mut outer_attrs = Vec::new();
+                    // let mut inner_attrs = Vec::new();
+                    // for attr in &data.attrs {
+                    //     match attr.style {
+                    //         crate::AttrStyle::Outer => outer_attrs.push(attr),
+                    //         crate::AttrStyle::Inner => inner_attrs.push(attr),
+                    //     }
+                    // }
+                    let (outer_attrs, inner_attrs): (Vec<_>, Vec<_>) = data
+                        .attrs
+                        .iter()
+                        .partition(|attr| matches!(attr.style, crate::AttrStyle::Outer));
```

Comment:
https://doc.rust-lang.org/stable/std/primitive.slice.html#method.partition_point can also be used here.